std/typecons.d

```diff
@@ -796,35 +796,71 @@ if (distinctFieldNames!(Specs))
          *           for comparison between `Tuple`s.
          *
          * Returns:
-         * For any values `v1` on the right-hand side and `v2` on the
-         * left-hand side:
+         * For values `v1` on the left-hand side and `v2` on the right-hand side:
+         *
+         * 0 if the expression `v1 == v2` is true for all members, or else for
+         * the first member, where this is not true:
          *
          * $(UL
-         *   $(LI A negative integer if the expression `v1 < v2` is true.)
-         *   $(LI A positive integer if the expression `v1 > v2` is true.)
-         *   $(LI 0 if the expression `v1 == v2` is true.))
+         *   $(LI NaN, in case one of the operands is a NaN.)
+         *   $(LI A negative number if the expression `v1 < v2` is true.)
+         *   $(LI A positive number if the expression `v1 > v2` is true.))
          */
-        int opCmp(R)(R rhs)
+        auto opCmp(R)(R rhs)
         if (areCompatibleTuples!(typeof(this), R, "<"))
         {
             static foreach (i; 0 .. Types.length)
             {
                 if (field[i] != rhs.field[i])
                 {
+                    import std.math : isNaN;
+                    static if (isFloatingPoint!(Types[i]) && isFloatingPoint!(typeof(rhs.field[i])))
```

Comment:
This is kinda redundant as both checks will be generated individually below.
(=> Smaller change)